exporter_test.go

```diff
@@ -910,156 +911,163 @@ version = "4.5.6"
 					})
 
-					it("sets CNB_PROCESS_TYPE", func() {
+					it("sets the ENTRYPOINT to this process type", func() {
 						_, err := exporter.Export(opts)
 						h.AssertNil(t, err)
 
-						val, err := fakeAppImage.Env("CNB_PROCESS_TYPE")
-						h.AssertNil(t, err)
-						h.AssertEq(t, val, "some-process-type")
+						checkEntrypoint(t, fakeAppImage, "process", "some-process-type")
 					})
 
-					it("sets ENTRYPOINT to launcher", func() {
+					it("doesn't set CNB_PROCESS_TYPE", func() {
 						_, err := exporter.Export(opts)
 						h.AssertNil(t, err)
 
-						ep, err := fakeAppImage.Entrypoint()
+						val, err := fakeAppImage.Env("CNB_PROCESS_TYPE")
 						h.AssertNil(t, err)
-						h.AssertEq(t, len(ep), 1)
-						if runtime.GOOS == "windows" {
-							h.AssertEq(t, ep[0], `c:\cnb\lifecycle\launcher.exe`)
-						} else {
-							h.AssertEq(t, ep[0], `/cnb/lifecycle/launcher`)
-						}
+						h.AssertEq(t, val, "")
 					})
+				})
 
-					when("default process type is not in metadata.toml", func() {
-						it("returns an error", func() {
-							opts.DefaultProcessType = "some-missing-process"
-							_, err := exporter.Export(opts)
-							h.AssertError(t, err, "default process type 'some-missing-process' not present in list [some-process-type]")
-						})
+				when("it is set to a process type that doesn't exist", func() {
+					it.Before(func() {
+						opts.DefaultProcessType = "some-non-existing-process-type"
+						h.RecursiveCopy(t, filepath.Join("testdata", "exporter", "default-process", "one-non-default-process", "layers"), opts.LayersDir)
+					})
+					it("fails", func() {
+						_, err := exporter.Export(opts)
+						h.AssertError(t, err, "tried to set some-non-existing-process-type to default but it doesn't exist")
 					})
 				})
+			})
 
-				when("platform API is >= 0.4", func() {
-					it("sets the ENTRYPOINT to the default process", func() {
-						opts.DefaultProcessType = "some-process-type"
+			when("-process-type is not set", func() {
+				when("there are no default=true processes in metadata.toml", func() {
+					it.Before(func() {
+						h.RecursiveCopy(t, filepath.Join("testdata", "exporter", "default-process", "one-non-default-process", "layers"), opts.LayersDir)
+					})
+					it("send an info message that there is no default process, and sets the ENTRYPOINT to the launcher", func() {
 						_, err := exporter.Export(opts)
 						h.AssertNil(t, err)
-
-						ep, err := fakeAppImage.Entrypoint()
-						h.AssertNil(t, err)
-						h.AssertEq(t, len(ep), 1)
-						if runtime.GOOS == "windows" {
-							h.AssertEq(t, ep[0], `c:\cnb\process\some-process-type.exe`)
-						} else {
-							h.AssertEq(t, ep[0], `/cnb/process/some-process-type`)
-						}
+						assertLogEntry(t, logHandler, "no default process type")
+						checkEntrypoint(t, fakeAppImage, "lifecycle", "launcher")
 					})
+				})
 
-					it("does not set CNB_PROCESS_TYPE", func() {
+				when("there is a default=true process in metadata.toml", func() {
```

Comment:
This context might also benefit from a `it("doesn't set CNB_PROCESS_TYPE"` expectation.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/commits/d7d5d23a49855cf3585cff952586e1e79213dc4a.